rsmtool/configuration_parser.py

```diff
@@ -1055,8 +1060,8 @@ def normalize_validate_and_process_config(self, context='rsmtool'):
         self._check_config_is_loaded()
 
         self.normalize_config()
-        self.validate_config(context=context)
         self.process_config()
+        self.validate_config(context=context)
```

Comment:
Why did we change the order of these two? Was it wrong before?

Reply:
I decided that `validate_config` was the best place to make sure the number of experiment names matches the number of experiments since this is where we check for other abnormalities in the config file.  To do this, I needed to first make sure both fields are correctly parsed as lists (`process_config`) which is why I switched the order of the two operations.

In fact, some of our existing checks in `validate_config` also implicitly assumed the correct typing of the fields. E.g. check #9 which looks at whether `predict_expected_scores` is set to `True` only for models that support it. Previously, if you supplied `{'model': 'NNLR', 'predict_expected_scores': 'false'}`,  `false` would be treated as  a string, `if new_config[predict_expected_scores]` would be interpreted as `True` and the check would fail since `NNLR` does not support `predict_expected_scores`.

This is a very edge case but it illustrates that the new order is better aligned with our logic. I added a few tests for this too.